src/components/ProgressBar/index.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useRef } from 'react';
+import { Animated, Easing, Text, View } from 'react-native';
+import { ProgressBarStyles } from './ProgressBarStyle';
+
+export type ProgressBarProps = {
+  progress: number;
+};
+
+const ProgressBarInternal = (props: ProgressBarProps) => {
+  const translateX = useRef(new Animated.Value(0));
+  const styles = ProgressBarStyles;
+
+  useEffect(() => {
+    Animated.timing(translateX.current, {
+      toValue: props.progress / 100,
+      easing: Easing.inOut(Easing.ease),
+      useNativeDriver: false,
```

Comment:
I have zero experience with react-native so i dont know how this works, just make sure to cleanup any effects if this needs to.

Reply:
nice point, add cleanup function